opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -337,14 +337,14 @@ namespace Opm
         }
 
         /// One current control per group.
-        void setCurrentInjectionGroupControl(const std::string& groupName, const Group::InjectionCMode& groupControl ) {
-            current_injection_group_controls_[groupName] = groupControl;
+        void setCurrentInjectionGroupControl(const Opm::Phase& phase, const std::string& groupName, const Group::InjectionCMode& groupControl ) {
+            current_injection_group_controls_[std::make_pair(phase, groupName)] = groupControl;
         }
-        const Group::InjectionCMode& currentInjectionGroupControl(const std::string& groupName) const {
-            auto it = current_injection_group_controls_.find(groupName);
+        const Group::InjectionCMode& currentInjectionGroupControl(const Opm::Phase& phase, const std::string& groupName) const {
+            auto it = current_injection_group_controls_.find(std::make_pair(phase, groupName));
 
             if (it == current_injection_group_controls_.end())
-                OPM_THROW(std::logic_error, "Could not find any control for injection group " << groupName);
+                OPM_THROW(std::logic_error, "Could not find any control for " << phase << " injection group " << groupName);
```

Comment:
The `phase` would be converted to an integer and not make much sense, we should get the corresponding string instead, also add a few words to make the message "Could not find any control for phase GAS for injection group INJ".

Reply:
Doesn't 
`std::ostream& operator<<( std::ostream&, const Phase& );` in Runspec.hpp
convert it to string? 

at least std::cout << phase  << std::endl; works perfectly.

Reply:
You are right, I did not know of that operator.